var/spack/repos/builtin/packages/libunwind/package.py

```diff
@@ -32,3 +34,22 @@ class Libunwind(AutotoolsPackage):
     url      = "http://download.savannah.gnu.org/releases/libunwind/libunwind-1.1.tar.gz"
 
     version('1.1', 'fb4ea2f6fbbe45bf032cd36e586883ce')
+
+    # On Darwin, libunwind is available without any additional include paths
+    # and libraries. In order to support that, we allow configuring libunwind
```

Comment:
what do you mean by `additional include paths and libraries`? Does macOS provide `Libunwind`? if so, it can be set as an external package and one would still need to link against some libs.

Reply:
That's the point, on Mac you do not need any command-line options to use the libunwind library - the header is in `/usr/include` and the there is no need to add `-lunwind` since `libc.dylib` is already linked with `/usr/lib/system/libunwind.dylib`.

Reply:
Ok, maybe mention that in the incode comment?

Reply:
To be fair, I don’t like this as it applies only to externally provided package. 

What happens if user type : spack install libunwind ? 

You kind of assume it’s setup as an externally provided package (in this case by system).

Reply:
One can check that self.prefix=usr/ in addition to being macOS,
but what if some package need full path to the lib at configuration stage? 

I still expect libs() to return something even if it’s unnecessary.

@alalazo what do you think?

Reply:
OK, I'll add more details to the comment.

My goal was to allow users to use the mac os provided libunwind by adding it to their packages.yaml with a fake prefix. If a user types, `spack install libunwind` (and the external libunwind is not set as `builadable: False`) then spack should build and install the preferred libunwind version and `libs` should pick it up as normal.

If a package needs the path to the actual file? I don't know - it really depends what the package wants to do with the actual path. The question in this case is: is `libs` required to always return at least one path to an existing file? Or empty LibraryList is OK?

Reply:
> The question in this case is: is libs required to always return at least one path to an existing file? 

indeed. If the installation is successful (including the case of externally provided package), my take on that is `libs()` should always return something, even if we don't need it for linking. But let's see what others have to say....

Reply:
@tgamblin ping.